### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by carolina-cza** [carolina-cza/Tecna#8] [Weekly Blog]
Die letze Woche hab ich mich zum beginn bisschen mit pygame beschäftigt und da ein, zwei Sachen programmiert, um mich bisschen an das Thema anzutasten(für unser Programmier Projekt).

Im weiteren hab ich mich auch mal an das Thema API rangetraut. Für alle die nicht wissen was API (=Application Programming Interface) ist, allgemein kann man sagen das ein Programmierschnittstelle und ist somit eine Zugangspunkte zwischen Systemen. 
Ein Beispiel wo das Anwendung findet sind Wetterapps, wo die Anbieter auf die selben Wetterdaten zugreifen aber sie dann in ihrem Programm verwenden und verschieden darstellen, ohne aber eine eigene Wetterstation zu besitzen.
Ich hab es etwas einfacher genutzt und hab auf eine Cocktail Seite zu gegriffen um random Rezepte für Cocktails als output zu bekommen ([RandomCocktailAPI](https://github.com/carolina-cza/Tecna/blob/codespace-super-space-yodel-x55ppqjx97jqfp699/daily_stuff/API/RandomCocktailAPI.py)) und dann hab ich das selbe nochmal mit Rezepten ausprobiert [Recipe](https://github.com/carolina-cza/Tecna/blob/codespace-super-space-yodel-x55ppqjx97jqfp699/daily_stuff/API/RecipeAPI.py) und es hat erstaunlich gut funktioniert. 
Die Seiten auf die ich mich bezogen habe sind for free und extra für so Spielereien gemacht, weil normaler Weise ist es nicht so leicht sich auf eine Website zu beziehen.
Eine gute Erklärung was denn APIs überhaupt sind könnt ihr hier finden: [API](https://youtu.be/ZveW4_ZJtVY?si=3ekT8AC7vPCSIA6Y)
(vermutlich findet ihr dort eine bessere und verständlichere Erklärung als was ich euch gegeben hab^^)

Bis zum nächsten mal^^

**Comment by Amalia-Dann** [comment]
Sehr cool, dass du APIs ausprobiert hast und auch auch sehr coole Codes. 
Aufjedenfall clever, dass du Übung für dein großes Projekt hast :)
